util/src/main/java/com/redhat/lightblue/util/ServoGraphiteSetup.java

```diff
@@ -98,31 +110,89 @@ private static synchronized void doInitialize() {
             }
         }
 
-        String host = System.getenv("GRAPHITE_HOSTNAME");
-        LOGGER.debug("GRAPHITE_HOSTNAME environment variable in doInitialize() is " + host);
-        String port = System.getenv("GRAPHITE_PORT");
-        LOGGER.debug("GRAPHITE_PORT environment variable in doInitialize() is " + port);
         if (port == null) {
             port = "2004"; //default graphite port
             LOGGER.debug("Using default port: " + port);
         }
 
         String addr = host + ":" + port;
-        MetricObserver observer = new GraphiteMetricObserver(prefix, addr);
+
+        LOGGER.debug("GraphiteMetricObserver prefix: " + prefix);
+        LOGGER.debug("GraphiteMetricObserver address: " + addr);
+
+        observers.add(new GraphiteMetricObserver(prefix, addr));
+    }
+
+    /**
+     * If there is sufficient configuration, register a StatsD metric observer
+     * to publish metrics. Requires at a minimum environment variable
+     * STATSD_HOSTNAME. Optionally can set STATSD_PREFIX as well as STATDS_PORT.
+     * STATSD_PREFIX defaults to an empty string. STATSD_PORT defaults to
+     * '8125'.
+     */
+    private static void registerStatsdMetricObserver(List<MetricObserver> observers) {
+        String prefix = System.getenv("STATSD_PREFIX");
+        String host = System.getenv("STATSD_HOSTNAME");
+        String port = System.getenv("STATSD_PORT");
+
+        // verify at least hostname is set, else cannot configure this observer
+        if (null == host || host.trim().isEmpty()) {
+            LOGGER.info("StatdsMetricObserver not configured, missing environment variable: STATSD_HOSTNAME");
+            return;
+        }
+
+        LOGGER.debug("STATSD_PREFIX environment variable is: " + prefix);
+        LOGGER.debug("STATSD_HOSTNAME environment variable is: " + host);
+        LOGGER.debug("STATSD_PORT environment variable is: " + port);
+
+        int iport = -1;
+
+        try {
+            iport = Integer.valueOf(port);
+        } catch (NumberFormatException e) {
+            iport = -1;
```

Comment:
Better log a message here as it might do something unexpected (for instance, if I initialized the environment variable  with a single letter in the end of it by mistake I could lose some time to figure out that the statd is working in a different port )